ethereumj-core/src/main/java/org/ethereum/net/server/EthereumChannelInitializer.java

```diff
@@ -57,11 +62,35 @@ public void initChannel(NioSocketChannel ch) throws Exception {
                 logger.debug("Open {} connection, channel: {}", isInbound() ? "inbound" : "outbound", ch.toString());
             }
 
-            if (isInbound() && channelManager.isRecentlyDisconnected(ch.remoteAddress().getAddress())) {
-                // avoid too frequent connection attempts
-                logger.debug("Drop connection - the same IP was disconnected recently, channel: {}", ch.toString());
-                ch.disconnect();
-                return;
+            // For incoming connection drop if..
+            if (isInbound()) {
+                boolean needToDrop = false;
+                // Bad remote address
+                if (ch.remoteAddress() == null) {
+                    logger.debug("Drop connection - bad remote address, channel: {}", ch.toString());
+                    needToDrop = true;
+                }
+                // Avoid too frequent connection attempts
+                if (!needToDrop && channelManager.isRecentlyDisconnected(ch.remoteAddress().getAddress())) {
+                    logger.debug("Drop connection - the same IP was disconnected recently, channel: {}", ch.toString());
+                    needToDrop = true;
+                }
+                // Drop bad peers before creating channel
+                if (!needToDrop && nodeManager.getNodeStatistics(new Node(new byte[0], ch.remoteAddress().getHostString(),
```

Comment:
Requesting for `nodeManager.isReputationPenalized(InetSocketAddress addr)` which does all the job